diff --git a/payment.module b/payment.module
index d4cd730..e5766c6 100644
--- a/payment.module
+++ b/payment.module
@@ -879,7 +879,8 @@ function payment_amount_human_readable($amount, $currency_code = NULL) {
   if (payment_currency_version() === 2) {
     $currency = currency_load($currency_code);
     if ($currency) {
-      return $currency->format($amount);
+      $round_amount = $currency->roundAmount($amount);
+      return $currency->format($round_amount);
     }
   }
   $amount_string = number_format($amount, 2, '.', '');
--
2.5.4 (Apple Git-61)

